pandas/core/computation/eval.py

```diff
@@ -373,7 +375,11 @@ def eval(
             # if returning a copy, copy only on the first assignment
             if not inplace and first_expr:
                 try:
-                    target = env.target.copy()
+                    target = env.target
+                    if isinstance(target, NDFrame):
+                        target = target.copy(deep=not using_copy_on_write())
```

Comment:
You could also do `target.copy(deep=None)` here, which will have the same effect (without having to import `using_copy_on_write` here). I think this is how we did it in most places that had a `self.copy()` in the method

Reply:
Forgot about that, thx